middleware/kubernetes/parse.go

```diff
@@ -101,6 +98,5 @@ func (r recordRequest) String() string {
 	s += "." + r.service
 	s += "." + r.namespace
 	s += "." + r.podOrSvc
-	s += "." + r.zone
```

Comment:
This function will escape to stack.  Consider using strings.Join() instead of "+"